File: src/main/java/com/saybetter/domain/review/domain/RecordSymbol.java
package com.saybetter.domain.review.domain;

import com.saybetter.domain.symbol.domain.Symbol;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Getter
@Entity(name = "RECORD_SYMBOL")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class RecordSymbol {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "rs_id")
	private Long rsId;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "record_id", nullable = false)
	private Record record;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "symbol_id", nullable = false)
	private Symbol symbol;

	@Column(name = "touch_count", nullable = false, columnDefinition = "int default 0")
	private Integer touchCount;

}

File: storage/storage-mysql-kt/src/main/kotlin/io/say/better/storage/mysql/domains/review/entity/RecordSymbol.kt
package io.say.better.storage.mysql.domains.review.entity

import io.say.better.storage.mysql.domains.symbol.entity.Symbol
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType.LAZY
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.time.LocalDateTime
import java.util.*

@Entity(name = "RECORD_SYMBOL")
class RecordSymbol(
    @Id
    @Column(name = "rs_id")
    private var rsId: UUID,
    @JoinColumn(name = "record_id", nullable = false)
    @ManyToOne(fetch = LAZY)
    private val recorded: Record,
    @JoinColumn(name = "symbol_id", nullable = false)
    @ManyToOne(fetch = LAZY)
    private val symbol: Symbol,
    @Column(name = "touch_order", nullable = false)
    private val touchOrder: Int,
    @Column(name = "touch_time", nullable = false)
    private val touchTime: LocalDateTime,
)
